extensions/amp-analytics/0.1/events.js

```diff
@@ -1202,6 +1232,20 @@ export class VisibilityTracker extends EventTracker {
     });
   }
 
+  /**
+   * Returns a Promise indicating that we're ready to report the analytics,
+   * in the case of reportWhen: endOfFrame
+   * @return {!Promise}
+   * @private
+   */
+  createReportReadyPromiseForEndOfFrame_() {
+    const deferred = new Deferred();
+    const root = this.root.getRoot();
+    root.addEventListener('unload', () => deferred.resolve());
```

Comment:
can we remove eventListener in the handler as well?

Reply:
Done, but please confirm if you like the way I did this.